src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -135,8 +135,8 @@
 #else
 #define DO_UPDATE_INSTRUCTION_COUNT(opcode)                                                          \
 {                                                                                                    \
-    BytecodeCounter::_counter_value++;                                                               \
-    BytecodeHistogram::_counters[(Bytecodes::Code)opcode]++;                                         \
+    if (CountBytecodes || TraceBytecodes || StopInterpreterAt) BytecodeCounter::_counter_value++;    \
```

Comment:
Elsewhere we do `[...] || StopInterpreterAt > 0` which is clearer as it's a flag expecting an int value. Perhaps you could use that instead of the implicit boolean conversion?

Yes, this doesn't seem consistent throughout the code, but while at it...

Reply:
Right on. I went on fixing the uses of `StopInterpreterAt` in this macro. I thought we can just check it against `_counter_value` without implicitly checking for `0` in the line below. But then Zero fastdebug build times exploded on me! So I thought to rewrite the block to hide most of the branches under the big one. See the update. This also clusters the uses of `BytecodeCounter::_counter_value`.

Reply:
Thanks for the update!